easy-rules-core/src/main/java/org/jeasy/rules/core/DefaultRulesEngine.java

```diff
@@ -217,29 +219,37 @@ private boolean shouldBeEvaluated(Rule rule, Facts facts) {
     }
 
     private void logEngineParameters() {
-        LOGGER.log(Level.INFO, "Rule priority threshold: {0}", parameters.getPriorityThreshold());
-        LOGGER.log(Level.INFO, "Skip on first applied rule: {0}", parameters.isSkipOnFirstAppliedRule());
-        LOGGER.log(Level.INFO, "Skip on first non triggered rule: {0}", parameters.isSkipOnFirstNonTriggeredRule());
-        LOGGER.log(Level.INFO, "Skip on first failed rule: {0}", parameters.isSkipOnFirstFailedRule());
-        LOGGER.log(Level.INFO, "Skip on missing fact: {0}", parameters.isSkipOnMissingFact());
+        LOGGER.info("Rule priority threshold: {}",
+            parameters.getPriorityThreshold());
+        LOGGER.info("Skip on first applied rule: {}",
+            parameters.isSkipOnFirstAppliedRule());
+        LOGGER.info("Skip on first non triggered rule: {}",
+            parameters.isSkipOnFirstNonTriggeredRule());
+        LOGGER.info("Skip on first failed rule: {}",
+            parameters.isSkipOnFirstFailedRule());
+        LOGGER.info("Skip on missing fact: {}",
+            parameters.isSkipOnMissingFact());
     }
 
     private void log(Rules rules) {
-        LOGGER.log(Level.INFO, "Registered rules:");
-        for (Rule rule : rules) {
-            LOGGER.log(Level.INFO, format("Rule { name = '%s', description = '%s', priority = '%s'}",
-                    rule.getName(), rule.getDescription(), rule.getPriority()));
+        if (LOGGER.isInfoEnabled()) {
```

Comment:
Do we really need this test? If the level is info, the message will be logged. If the level is higher than info, the message will not be logged automatically. Or I'm missing something?

Reply:
This check avoids run the for. If the logger is not active, no reason to iterate.